### PR TITLE
[CALCITE-6189] Improve FormatElementEnumTest

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
Improve FormatElementEnumTest make code clean.
https://issues.apache.org/jira/browse/CALCITE-6189

### CODE REVIEW

**Comment by caicancai** [comment]
Thank you

**Comment by julianhyde** [comment]
But is it an improvement? The test is more difficult to modify and understand.

See https://dev.to/mbarzeev/why-practicing-dry-in-tests-is-bad-for-you-j7f

**Comment by caicancai** [comment]
> But is it an improvement? The test is more difficult to modify and understand.
> 
> See https://dev.to/mbarzeev/why-practicing-dry-in-tests-is-bad-for-you-j7f

The test I wrote before may have some parts that can be optimized. There are some code redundancies that may be optimized.

**Comment by XuQianJin-Stars** [comment]
> But is it an improvement? The test is more difficult to modify and understand.
> 
> See https://dev.to/mbarzeev/why-practicing-dry-in-tests-is-bad-for-you-j7f

This pr has been updated.

File: core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java [hunk 1]
```diff
@@ -28,150 +28,87 @@
...
   }
-
   @Test void testFF1() {
-    final String date = "2014-09-30T10:00:00.123456Z";
-    StringBuilder ts = new StringBuilder();
-    FormatElementEnum.FF1.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("123"));
+    assertFormatElement(FormatElementEnum.FF1, "2014-09-30T10:00:00.123456Z", "123");
   }
-
```
Comment (tanclary) [inline core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java:72]: I don't think we need to remove the line breaks between tests.

**Review by tanclary — CHANGES_REQUESTED** [review]
One style comment otherwise LGTM

**Review by tanclary — CHANGES_REQUESTED** [review]
It still looks like there are removed lines between the tests

**Review by tanclary — APPROVED** [review]
I think this looks good. Would you mind squashing your commits? If you wanted to leave a more descriptive commit message, you could.

**Comment by caicancai** [comment]
@XuQianJin-Stars Good to merge?

File: core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java [hunk 2]
```diff
@@ -28,150 +28,94 @@
...
   @Test void testYYYY() {
-    final String date = "2014-09-30T10:00:00Z";
+    assertFormatElement(FormatElementEnum.YYYY, "2014-09-30T10:00:00Z", "2014");
+  }
+
+  private void assertFormatElement(FormatElementEnum formatElement, String date, String expected) {
     StringBuilder ts = new StringBuilder();
-    FormatElementEnum.YYYY.format(ts, Date.from(Instant.parse(date)));
-    assertThat(ts, hasToString("2014"));
+    formatElement.format(ts, Date.from(Instant.parse(date)));
```
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java:118]: I noticed that other test methods are placed above. Do we need to comply with this specification?

Comment (caicancai) [reply core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java:118]: Well, it seems irrelevant.

**Pull request merged**